src/openrct2/world/park.c

```diff
@@ -477,14 +477,18 @@ static sint32 park_calculate_guest_generation_probability()
 	return probability;
 }
 
+/**
+ * Choose a random peep spawn and iterates through until defined spawn is found.
+ */
 static void get_random_peep_spawn(rct2_peep_spawn *spawn)
 {
-	*spawn = gPeepSpawns[0];
-	if (gPeepSpawns[1].x != PEEP_SPAWN_UNDEFINED) {
-		if (scenario_rand() & 0x80000) {
-			*spawn = gPeepSpawns[1];
-		}
+	sint32 i = scenario_rand() % MAX_PEEP_SPAWNS;
+	for (uint8 tries = 0; tries < MAX_PEEP_SPAWNS; tries++) {
+		if (gPeepSpawns[i].x != PEEP_SPAWN_UNDEFINED)
+			break;
+		i = (i + 1) % MAX_PEEP_SPAWNS;
```

Comment:
This won't devide the peeps over multiple spawn points evenly. Suppose MAX_PEEP_SPAWNS is 10 and peep spawn 1 and 2 are defined, then 90% of the peeps will start at 1.